packages/conform-dom/form.ts

```diff
@@ -519,7 +519,7 @@ function createConstraintProxy(
 					// This overrides the current number segment with an empty string
 					// which will be treated as an empty bracket
 					path[i] = '';
-					break;
+					continue;
```

Comment:
I just had another look and I am not sure why this change fixes the issue actually.

The original `break` was intentional. When resolving a constraint we progressively generalise the field name and scan each alternative in order:

```
array[0].anotherArray[1].key
→ array[0].anotherArray[].key
→ array[].anotherArray[].key
```

Switching to `continue` skips the intermediate variant and jumps straight from the fully specific name to the fully general one here. 🤔

Reply:
Hmmm, just did a quick test with the `createStateProxy` and it seems to be working as expected:

```js
function createStateProxy(fn) {
	const cache = {};
	return new Proxy(cache, {
		get(_, name, receiver) {
			if (typeof name !== 'string') {
				return;
			}

			return (cache[name] ??= fn(name, receiver));
		},
	});
}

let t = createStateProxy((name, proxy) => {
  console.log('Looking up', name);
  if (name) {
    return proxy[name.slice(0, -1)];
  }

  console.log('done');
});

t.abc

// > Looking up abc
// > Looking up ab
// > Looking up a
// > Looking up 
// > done
```

Reply:
Thanks for looking into it.
I didn't realize it was fixed in #983 . Let's close this PR then.